File: Java/src/Dequeue/Dequeue.java
package Dequeue;

import java.util.ArrayList;

/**
 * Created by Daniel on 07.04.2015.
 */
public class Dequeue<T> {
    private final ArrayList<T> data = new ArrayList<>();
    public int size() {
        return data.size();
    }

    public void addFront(T item) {
        data.add(0, item);
    }

    public T removeRear() {
        return data.remove(data.size() - 1);
    }

    public void addRear(T item) {
        data.add(item);
    }

    public T removeFront() {
        return data.remove(0);
    }
}

File: Go/Dequeue/dequeue.go
package Dequeue


type Dequeue struct {
    data []interface{}
}

func (d *Dequeue) size() int {
    return len(d.data)
}

func (d *Dequeue) addFront(item interface{}) {
    d.data = append([]interface{}{item}, d.data...)
}

func (d *Dequeue) addRear(item interface{}) {
    d.data = append(d.data, item)
}

func (d *Dequeue) removeRear() interface{} {
    res := d.data[len(d.data) - 1]
    d.data = d.data[0:len(d.data) - 2]
    return res
}

func (d *Dequeue) removeFront() interface{} {
    res := d.data[0]
    d.data = d.data[1:]
    return res
}